File: 4045-longest-balanced-subarray-i/4045-longest-balanced-subarray-i.java
class Solution {
    public int longestBalanced(int[] nums) {
        int n = nums.length;
        int maxi = 0;
        for (int i = 0; i < n; i++) {
            HashSet<Integer> even = new HashSet<>();
            HashSet<Integer> odd = new HashSet<>();
            for (int j = i; j < n; j++) {
                if (nums[j] % 2 == 0)
                    even.add(nums[j]); 
                else 
                    odd.add(nums[j]);
                if (even.size() == odd.size()) {
                    maxi = Math.max(maxi, j - i + 1);
                } 
            }
        }
        return maxi; 
    }
}

File: 4045-longest-balanced-subarray-i/4045-longest-balanced-subarray-i.py
class Solution:
    def longestBalanced(self, nums: List[int]) -> int:
        n = len(nums)
        maxi = 0
        for i in range(0, n):
            odd_set = set()
            even_set = set()
            for j in range(i, n): 
                if nums[j] % 2 == 0: 
                    even_set.add(nums[j])
                else: 
                    odd_set.add(nums[j])
                if len(even_set) == len(odd_set): 
                    maxi = max(maxi, j - i + 1)   
        return maxi 